foreshadow/tests/test_core/test_fs_logging.py

```diff
@@ -0,0 +1,221 @@
+from unittest import mock
+
+import pytest
+
+
+@pytest.mark.parametrize("n_gets", [1, 2, 100])
+def test_get_logger(n_gets):
+    """Test the correct functionality of get_logger.
+
+    Args:
+        n_gets: number of times to try getting a logger.
+
+    """
+    from foreshadow.core import fs_logging
+    import logging as logging
+
+    logger = fs_logging.get_logger()
+    if n_gets == 1:
+        assert isinstance(logger, logging._loggerClass)
+    elif n_gets > 1:
+        for _ in range(n_gets - 1):
+            logger_ = fs_logging.get_logger()
+        assert logger == logger_
```

Comment:
This is just a suggestion (I know we chatted about this) I would prefer if we try to break up logging statements that have if statements into separate tests. If there is a concern about setup and teardown (shared code), we can definitely use fixtures where necessary.

Reply:
will fix